add command line argument directio (#1943)

cclhsu opened the pull request:
#### Proposed Changes ####

add command line argument directio to resolve ticket '[BUG] sparse-tools: ssync should explicitly asking for directIO or not, rather than determine is using filesize' (#1943)

1. Add CLI argument `directIO` instead of set directIO using `filesize`
2. Update/Add unit test cases using `directIO`

#### Types of Changes ####

Bug fixing.

#### Verification ####
 
Updated/Added unit test cases

#### Linked Issues ####

Refs: # https://github.com/longhorn/longhorn/issues/1943

#### Further Comments ####

None

Signed-off-by: cclhsu <clark.hsu@suse.com>

shuo-wu reviewed (changes requested):
1. Should we disable the DiectIO by default? I think we need to enable it by default.
2. Remember to update the vendor for repo `longhorn/longhorn-engine`, as well as this: https://github.com/longhorn/longhorn-engine/blob/1481ab6dd315557d10c7e51081b966b39d694ed0/pkg/sync/rpc/server.go#L333

yasker commented:
@cclhsu Is this PR ready to be reviewed/merged? Remember to ping the reviewer or re-request the review if the PR is ready for review after the update. Otherwise, it's hard to tell from the reviewer's perspective if the PR is ready for re-review or not.

cclhsu commented:
@shuo-wu I have update the PR and https://github.com/longhorn/longhorn-engine/pull/560 is ready for review after this ticket is merged.

shuo-wu reviewed (changes requested):
Squash this commit into the 1st one. Then this PR LGTM

cclhsu commented:
> Squash this commit into the 1st one. Then this PR LGTM

@shuo-wu Do you mean the function in [Allow squash merging](https://docs.github.com/en/github/administering-a-repository/configuring-commit-squashing-for-pull-requests)? or manually reset to head and squash the patches?

shuo-wu commented:
I mean coalescing the 2nd commit into the 1st commit. 
There is an option named `squash` can help you do that when you run `git rebase -i origin/master`. Hence I use word `squash` as an abbreviation.

Commit:
add command line argument directio (#1943)

add command line argument directio to resolve ticket '[BUG] sparse-tools: ssync should explicitly asking for directIO or not, rather than determine is using filesize' (#1943)

Signed-off-by: Clark Hsu <clark.hsu@suse.com>

shuo-wu commented:
I think `directIO` in old test cases should be true?

cclhsu commented:
> I think `directIO` in old test cases should be true?

For ssync client file, I have `directIO` as set false since it is testing existing file. 
For ssync random file, I created new test cases to test `directIO` as true and false.

shuo-wu commented:
OK, I see. For those existing tests, the files the tests generate are not multiple of 512 hence the flag should be false
